### ISSUE TITLE
[CI Problem] Flaky Test: `tests/python/unittest/test_meta_schedule_schedule_rule_auto_inline.py::test_inline_into_cache`

### ISSUE THREAD

**Issue by junrushao1994** [apache/tvm#9952]
### Branch/PR Failing

Looks like [this PR](https://github.com/apache/tvm/pull/9943) introduced a flaky test. Likely it's due to some concurrent merge.

### Jenkins Link

https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/main/2417/pipeline/

### Flakiness

Yes.

CC: @MasterJH5574

**Comment by junrushao1994** [comment]
CC @wrongtest @Hzfengsy

It looks like some randomness was introduced recently in `GetBlockReadWriteRegion`, possibly in [this commit](https://github.com/apache/tvm/commit/6f6fc68f5a028a92607b2907b9e4144543686639), but this particular robustness issue wasn't revealed until @MasterJH5574's PR:

![image](https://user-images.githubusercontent.com/22515877/149885366-a3b1d582-dd92-4984-9f59-82978636d62a.png)

**Comment by wrongtest** [comment]
cc @junrushao1994 @Hzfengsy  

The fix trial is here https://github.com/apache/tvm/pull/9958

I am sorry for introducing this issue. I believe it is introduced by https://github.com/apache/tvm/pull/9372, but detected until subsequent change https://github.com/apache/tvm/pull/9880 which touch more testcases.

Developers have already pointed out that with equations like `xx + rx >= 1`, the `SolveInequalitiesToRange` create undetermined results for the failed case. After a look through I found that actually the result of ` SolveInequalitiesToRange` depend on the to-solve variables order we pass in. However, in the region analysis context, I used an unordered_dict with ObjectPtrHash to record the variables.
